### PR TITLE
Callback context

### PR DESCRIPTION
**Pull request by waralex**
Added the `callback_context()` function. This function only works from callback processing functions and returns a CallbackContext object :
```julia
mutable struct CallbackContext
    response::HTTP.Response
    inputs::Dict{String, Any}
    states::Dict{String, Any}
    outputs_list::Vector{Any}
    inputs_list::Vector{Any}
    states_list::Vector{Any}
    triggered::Union{Nothing, Vector{TriggeredParam}}
end
```

In addition, I refactored the code structure - distributed independent parts of the code to different files, designed the routing-related functionality as a sub-module, and so on. Therefore, the number of changed files in PR turned out to be significant, but for the most part it is copying code from old files to new ones.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       - run:
           name: 🔎 Unit tests
           command: |
-            julia -e 'using Pkg; Pkg.update(); Pkg.add(PackageSpec(url="https://github.com/plotly/Dash.jl.git", rev=ENV["CIRCLE_BRANCH"])); Pkg.add(PackageSpec(url="https://github.com/waralex/dash-html-components.git", rev="jl_generator_test")); Pkg.add(PackageSpec(url="https://github.com/waralex/dash-core-components.git", rev="jl_generator_test")); Pkg.build("Dash"); Pkg.build("DashHtmlComponents"); Pkg.build("DashCoreComponents"); Pkg.test("Dash", coverage=true); function precompile_pkgs(); for pkg in collect(keys(Pkg.installed())); if !isdefined(Symbol(pkg), :Symbol) && pkg != "Compat.jl"; @info("Importing $(pkg)..."); try (@eval import $(Symbol(pkg))) catch end; end; end; end; precompile_pkgs()'
+            julia test/ci_prepare.jl
```
Comment (alexcjohnson) [inline .circleci/config.yml:31]: 😎 🍹 Much nicer, thanks!

File: src/handler/callback_context.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+const CallbackContextItems = Union{Nothing, Vector{NamedTuple}}
+const TriggeredParam = NamedTuple{(:prop_id, :value)}
+mutable struct CallbackContext
+    response::HTTP.Response
+    inputs::Dict{String, Any}
+    states::Dict{String, Any}
+    outputs_list::Vector{Any}
+    inputs_list::Vector{Any}
+    states_list::Vector{Any}
+    triggered::Union{Nothing, Vector{TriggeredParam}}
```
Comment (alexcjohnson) [inline src/handler/callback_context.jl:11]: Is the `Nothing` necessary? Won't `triggered` at least be a length-0 Vector? Just trying to make sure this is as easy to use as possible, so users don't need fallback code for the `Nothing` case.

Comment (waralex) [reply src/handler/callback_context.jl:11]: @alexcjohnson I agree, I'll fix it

File: src/handlers.jl [hunk 3]
```diff
@@ -1,5 +1,7 @@
-include("handler/router.jl")
-include("handler/base.jl")
+#include("handler/router.jl")
+#include("handler/base.jl")
```
Comment (alexcjohnson) [inline src/handlers.jl:2]: obsolete?

File: src/handler/callback_context.jl (hunk 2 above)
Comment (waralex) [reply src/handler/callback_context.jl]: @alexcjohnson fixed at eb51d4b

File: src/handlers.jl (hunk 3 above)
Comment (waralex) [reply src/handlers.jl]: @alexcjohnson fixed at eb51d4b

**Pull request merged**